task/fbc-validation/0.1/fbc-validation.yaml

```diff
@@ -0,0 +1,76 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  labels:
+    app.kubernetes.io/version: "0.1"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: "appstudio, hacbs"
+  name: fbc-validation
+spec:
+  params:
+    - name: IMAGE_URL
+      description: the fully qualified image name
+  results:
+    - name: HACBS_TEST_OUTPUT
+  steps:
+    - name: check-labels-and-binaries
+      image: quay.io/redhat-appstudio/hacbs-test:latest
+      workingDir: /work
+      securityContext:
+        runAsUser: 1000
+      env:
+        - name: HOME
+          value: /work
+      resources:
+        limits:
+          memory: 4Gi
+          cpu: 2
+        requests:
+          memory: 512Mi
+          cpu: 10m
+      script: |
+        ### Try to extract binaries with configs > check binaries functionality > check opm validate ###
+
+        mkdir confdir
+        if ! oc image extract $(params.IMAGE_URL) --file /bin/opm --file /bin/grpc_health_probe --path configs/*:confdir/ ; then
+        echo "Unable to extract image! Skipping checking binaries!"
+          HACBS_ERROR_OUTPUT=$(jq -rc --arg date $(date +%s) --null-input \
+          '{result: "FAILURE", timestamp: $date, successes: 0, failures: 1}')
+          echo "${HACBS_TEST_OUTPUT:-${HACBS_ERROR_OUTPUT}}" | tee $(results.HACBS_TEST_OUTPUT.path)
+          exit 0
+        fi
+
+        TESTPASSED=true
+        chmod +x opm grpc_health_probe
+
+        if ! ./opm version; then
+        echo "!FAILURE! - opm binary check failed"
+          #lets update the reuslt here??
+          TESTPASSED=false
+        fi
+        if [ ! -f "grpc_health_probe" ]; then
+        echo "!FAILURE! - grpc_health_probe binary check failed"
+          #lets update the reuslt here??
+          TESTPASSED=false
+        fi
+        if ! ./opm validate confdir; then
```

Comment:
I'm not sure about using the `opm` binary from the image under test to self-validate it. It could be the way to go, but it might be best to consult the operator framework folk about how we should approach this.

The other way would be to package a specific version of `opm` inside our `quay.io/redhat-appstudio/hacbs-test:latest` image that's used in this task and use that for validation instead.

Reply:
what I understood it's even sensitive on openshift version, like when it's for 4.11 it should be version xxx and for 4.10 version yyy

Reply:
Created a story that should take care of such cases:
https://issues.redhat.com/browse/HACBS-1567
FCB-validate task will be modified accordingly afterwards.

Reply:
Ok, that makes sense, thanks. We can keep the code as-us until the follow up story is worked out.